atsign-foundation/at_app_cli
# Auto generated at_template tooling

Please review and merge #66 first.

**- What I did**

`at_app_templates/` & `templates/`

- Discontinued at_app_templates package, renamed directory to templates.

`tools/`
- Wrote a tool to automatically generate template code for templates (template_bundler).
This tool calls `mason bundle -d dart` on the file.  Then it generates the dart code for the at_template, using `template.yaml` to configure it.

- Updated build scripts to point to `templates` instead of `at_app_templates`

`at_app/`

- Rebundled at_app using the template_bundler tool mentioned above.
- Removed old template code
- Updated import references and `templates.dart`

`melos.yaml`
Updated package paths and script names.

**- How I did it**

Local changes

**- How to verify it**

This PR contains all of the changes from #66, verifying this will also verify it.

1. Pull this branch
2. Bootstrap with melos: `dart pub global run melos bootstrap`
3. Run the melos build scripts to verify the templates were built successfully: `dart pub global run melos run build_aat`
4. Run at_app locally and verify that the project generated correctly
ex. `dart run at_app/bin/at_app.dart create -d snackbar testsnack`
5. Verify that the output app is correct.
  - correct `lib/main.dart`
  - correct `.env`

.env content should be:
```
NAMESPACE=fourballcorporate9
API_KEY=477b-876u-bcez-c42z-6a3d
```

**- Description for the changelog**
Auto generated at_template tooling
